### ISSUE TITLE
Full resolution image URLs are broken again

### ISSUE THREAD

**Issue by VasiliPupkin256** [rarcega/instagram-scraper#193]
The hack I used to fix `get_original_image` in #157 stop working.
Any ideas how to fix it again and get the full resolution image URL?

**Comment by mattchete** [comment]
In Instagram JSON, the full resolution image url is available at the key "display_url".
example: "display_url": "https://scontent-cdt1-1.cdninstagram.com/vp/544bf247f8f97c57fc90d54814d7c23d/5B382779/t51.2885-15/e35/29087780_289722998226057_5244196321576878080_n.jpg",

**Comment by VasiliPupkin256** [comment]
@mattchete does it work for this image http://www.instagram.com/p/BS1colrjDMO/ ?

**Comment by lynxrv21** [comment]
@VasiliPupkin256 As a workaround, or a quick fix, maybe we can use `"display_url"` where it is available, and the largest one of `"thumbnail_resources"` if not

**Comment by VasiliPupkin256** [comment]
@lynxrv21 it looks like `display_url` has the biggest resolution among other resized, may not always be true. If so we can disable `get_original_image`, until someone finds a better way to get a full resolution image url

**Comment by lynxrv21** [comment]
@VasiliPupkin256 you mean `thumbnail_resources` can be smaller than one of `thumbnail_resources` ?
I can't see better solution for now, so can you implement this or similar logic? I mean get one of resized images instead of `get_original_image`

**Comment by VasiliPupkin256** [comment]
@lynxrv21 the other question is what to do with profile pic. `profile_pic_url_hd` is always pointing to a very small image 320x320

**Comment by lynxrv21** [comment]
@VasiliPupkin256 have no idea :) Have it been large before? Anyway, 320x320 is better than nothing (access denied in our case)

**Comment by VasiliPupkin256** [comment]
The key is to find any app or site or API interface project that is able to download a full resolution profile picture now

**Comment by rarcega** [comment]
As a stopgap solution I've resorted to leaving the media image and profile image urls unmodified in v1.5.25.

**Comment by Hrxn** [comment]
What about `og:image` metadata?
I.E. `<meta property="og:image"  ... `

IG definitely made use of this in the past..
Or is `display_url` the better choice? I have no idea btw., never really used IG in any way or form.

**Comment by ProfessorZer0** [comment]
https://play.google.com/store/apps/details?id=in.galaxyapps.storysaver
This app seemed to have fixed the issue.if anyone can fix it?

**Comment by VasiliPupkin256** [comment]
Interesting. It is getting profile pic from `https://i.instagram.com/api/v1/users/{ID}/info/` url. For example for lelepons with her id 177402262 we go to https://i.instagram.com/api/v1/users/177402262/info/ and voila here is our full resolution profilepic https://scontent-frx5-1.cdninstagram.com/vp/89295892651786d426a6c00c3372e5a4/5B379562/t51.2885-19/22280108_1622292011160332_5137192580106158080_n.jpg
I am not sure if it is possible to get profile pic bigger than 1080x1080 this way

**Comment by Hrxn** [comment]
But does it work for non profile pics?

**Comment by xC0dex** [comment]
@VasiliPupkin256 thank you for your answer. No, profile pictures have a maximum size of 1080x1080. Are you sure that we get always the full size image? How is the parameter called? "profile_pic_url"? :)

**Comment by VasiliPupkin256** [comment]
@Hrxn I've tried `.../media/{media_id}/info` requests through the [Instagram-API-python](http://github.com/LevPasha/Instagram-API-python) project, but for me it always return url with resolution worse than 1080x1080, don't know what I miss

**Comment by fantomnotabene** [comment]
@xC0dex jsonObject["user"]["hd_profile_pic_url_info"]["url"]

**Comment by xC0dex** [comment]
@fantomnotabene Thanks! So I am forced to use the API with an access_token "https://www.instagram.com/mkbhd/?__a=1" This public API is useless to get a full size image.

**Comment by fantomnotabene** [comment]
> "https://www.instagram.com/mkbhd/?__a=1"

is useful to obtain profile's identifier.
Its `response.json()['graphql']['user']['id']` takes you to the target.
@xC0dex then "https://i.instagram.com/api/v1/users/{profile_id}/info/" can be used with no access_token.

**Comment by xC0dex** [comment]
@fantomnotabene you are awesome! Thanks a lot! This is exactly what I'm looking for! Thumbs up

**Comment by MintUser34827738947** [comment]
I'm only getting 320x320 profile pictures on all my users I try to scrape. Some are private profiles I have no further access, some are private profiles I have access (I follow) and most of them are public profiles. What am I doing wrong? Thought the fix 7 days ago should help in that area? Using -u and -p of course

**Comment by VasiliPupkin256** [comment]
Yep. The patch is bad. `profile_pic_url` wasn't used as download url in `get_profile_pic` function

**Comment by tsrnfy** [comment]
"https://i.instagram.com/api/v1/users/{profile_id}/info/" is not working anymore.
Anyone to confirm that?

**Comment by mattchete** [comment]
Yup, endpoint result is an html page which asks for signing.
Anyone has come up with a solution ?

**Comment by mattchete** [comment]
A solution from the PHP version:
https://github.com/postaddictme/instagram-php-scraper/issues/433#issuecomment-445533330

**Comment by singhakash009** [comment]
anyone has found the solution?

**Comment by VasiliPupkin256** [comment]
There is a site http://instadp.com that somehow manages to get full resolution photos

**Comment by wbrickner** [comment]
`https://i.instagram.com/api/v1/users/{profile_ID}/` currently works.
Mind the fact that you must provide the user's *platform ID*, not their username.  It's a sequence of numbers.  You can extract it from the JSON inside a script tag embedded in the profile page, although it becomes a headache.

**Comment by Nakuls426** [comment]
Did Anyone got the solution because I'm also looking for the same any help will be appreciated.

**Comment by saidattax** [comment]
@a

> There is a site http://instadp.com that somehow manages to get full resolution photos

Any idea how they do this?

**Comment by Nakuls426** [comment]
> 
> > There is a site http://instadp.com that somehow manages to get full resolution photos
> 
> Any idea how they do this? 

Instadp uses cookies to get full resolution pics.

**Comment by silverappsstudios** [comment]
@Nakuls426 any idea which API do they pull the data from? 

> https://www.instagram.com/mkbhd/?__a=1

This one? This only gives me only 320, 640 

I'm looking for full size
